### DISCUSSION TITLE
iFrame Content to help Share in Contextualization of your site

### DISCUSSION THREAD

**Discussion by AdamFermier** [McShelby/hugo-theme-relearn#561] [Ideas]
I often find myself in a situation where I simply want to iframe a pdf, another HTML page, what ever - so have developed these shortcodes which I leveraged from friends who came before me.  Would it be useful to include for the 'relearn' theme?

### PDF as iFramed Content

filename= `iframe-pdf.html`
Content = `<embed src= "{{ .Get 0 }}" width= "100%" height= "1000px" type="application/pdf" >`

### html as Iframed Content

filename = `iframe-html.html`
Content = 
```
<script>
    function resizeIframe(obj) {
      obj.style.height =  1.05*obj.contentWindow.document.body.scrollHeight + 'px';
    }
  </script>

<iframe 
	height= 1250 px
	width= 1500 px
	name="iframe" 
	src="{{ index .Params 0 }}" 
    frameborder="0" 
    scrolling="yes" 
    position: relative
	allowfullscreen = TRUE
    onload="resizeIframe(this)">
</iframe>
```
### Youtube start and end

filename = `youtubestartend.html`
Content = 
```
<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
    <iframe src="https://www.youtube-nocookie.com/embed/{{ index .Params 0 }}?start={{ index .Params 1 }}&end={{ index .Params 2}}"
    style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" allowfullscreen frameborder="0" title="YouTube Video"></iframe>
</div>
```

**Comment by McShelby** [comment]
I am willing to have a general `iframe` shortcode but find it difficult to assess all the attributes and decide which to have configurable, which to be set and to what value. Often the use case requires certain configurations but I am not very experienced in using iframes.

The `embed` solution quite suffers from my same lack of experience but also seems a lot more limited while the use case of an embedded PDF is quite common and most valuable to theme users.

I will definetly not support another `youtube` shortcode. In addition to the above reasons that also apply here, there is already a `youtube` shortcode contained in Hugos distribution that can be readily used.

Additionally I will not include anything in the theme that a) isn't under an open source license b) solely depends on external sources.

With any of those proposed new shortcodes comes additional work in the form of making them work on mobile and print, which both could be quite a hassle.

**Reply by AdamFermier** [reply]
Thanks - no need to add them if it does not align - they are here if folks want to use them experimentally. Thanks for the feedback!
